### PR TITLE
Monitor test coverage

### PR DESCRIPTION
**Pull request by Rowlando13**
Added Coverage to our tool chain. It is setup to be able to run locally. Because of the time to execute ~1 min, it is not a good candidate for precommit. It should be integrated in our CI . Will update the docs to show how to use it on next pr.

### CODE REVIEW

**Comment by Rowlando13** [comment]
coverage usage

**Comment by Rowlando13** [comment]
Sample coverage report ^^^^

**Comment by nreinicke** [comment]
Nice, I like the idea of having these report to help us improve test coverage. Would you mind just adding the docs update to this PR?

**Comment by Rowlando13** [comment]
You got it.

**Comment by Rowlando13** [comment]
Updated the docs for coverage.

File: contributor_environment.yml [hunk 1]
```diff
@@ -15,3 +15,8 @@ dependencies:
   - black
   - isort
   - interrogate
+  - coverage
+
+
+=======
```
Comment (nreinicke) [inline contributor_environment.yml:21]: looks like this might be an artifact from a merge conflict resolution?

**Review by nreinicke — COMMENTED** [review]
Thanks for updating the docs, I left one small comment about some left over code from a merge

File: contributor_environment.yml (hunk 1 above)
Comment (Rowlando13) [reply contributor_environment.yml:21]: I thought that someone else put that in. I will remove it.

**Review by Rowlando13 — COMMENTED** [review]
Changes made and reviewed.

**Review by nreinicke — APPROVED** [review]
This looks great, thanks for adding this in

**Pull request merged**